src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,37 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  selectedTodo?: Todo;
+  setSelectTodoId: (selectTodo: number | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = props => {
+  const { selectedTodo, setSelectTodoId } = props;
+
+  const [loader, setLoader] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+  const userId = selectedTodo?.userId;
+
+  useEffect(() => {
+    setLoader(true);
+
+    if (userId) {
+      getUser(userId)
+        .then(data => setUser(data))
+        .catch()
```

Comment:
The `catch` block is empty. It's important to handle errors properly to avoid unhandled promise rejections. Consider logging the error or displaying an error message to the user.